core-java-modules/core-java-string-operations-4/src/test/java/com/baeldung/charcomparison/CharComparisonUnitTest.java

```diff
@@ -0,0 +1,65 @@
+package com.baeldung.charcomparison;
+
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import java.util.Objects;
+
+import org.junit.jupiter.api.Test;
+
+public class CharComparisonUnitTest {
+
+    @Test
+    public void givenTwoPrimitiveChars_whenUsingRelationalOperators_thenCompare() {
+
+        assertFalse('a' == 'A');
+        assertTrue('a' < 'v');
+        assertTrue('F' > 'D');
+
+    }
+
+    @Test
+    public void givenTwoPrimitiveChars_whenUsingCompareMethod_thenCompare() {
+
+        assertTrue(Character.compare('C', 'C') == 0);
+        assertTrue(Character.compare('f', 'A') > 0);
+        assertTrue(Character.compare('Y', 'z') < 0);
+
+    }
+
+    @Test
+    public void givenTwoCharObjects_whenUsingCompareToMethod_thenCompare() {
+
+        Character chK = Character.valueOf('K');
+        Character chG = Character.valueOf('G');
+        Character chH = Character.valueOf('H');
```

Comment:
The boxing is not really needed, you can just do `Character chK = 'K';`

Reply:
Yes I know, the idea here is to mention in the article that the constructor of the Character class is deprecated since Java 9
However, if you think that the information is irrelevant, please let me know ;)

Reply:
It´s OK to mention that info. With just one test we can show that case and having the rest with a direct initialization.

Reply:
Done